### ISSUE TITLE
Generating too many tempfiles

### ISSUE THREAD

**Issue by rtkkdeng** [hashicorp/vagrant#4103]
It seems that from running any vagrant command, my $TMPDIR gets overflowed with  vagrantYYYYMMDD-hmmss-XXXXXXX files and dYYYYMMDD-hmmss-XXXXXXX directories.

From vagrant/lib/bundler.rb, it creates these files every time I run a vagrant command and more depending on what vagrant command I run.
ie. Running 'vagrant --version' creates several of these tempfiles after each run.

Is there any reason why vagrant doesn't clean up these files after?

**Comment by caljess599** [comment]
I am experiencing this as well.

**Comment by varac** [comment]
same here, ubuntu 14.10, vagrant 1.7.1 debian package.

```
--- /tmp » ls /tmp/vagrant* | wc -l
1966

```

**Comment by varac** [comment]
upgraded to 1.7.2, removed all leftovers from /tmp, run "vagrant", and these files were put in /tmp:
```
--- /tmp » ls -al vagrant* d2015* -d
drwx------ 2 varac varac 4096 Feb  7 18:42 d20150207-30612-401o24
-rw------- 1 varac varac    0 Feb  7 18:42 vagrant20150207-30612-1htp2k3
-rw------- 1 varac varac    0 Feb  7 18:42 vagrant20150207-30612-owies3
-rw-rw-r-- 1 varac varac  289 Feb  7 18:42 vagrant20150207-30612-owies32
-rw-rw-r-- 1 varac varac 4324 Feb  7 18:42 vagrant20150207-30612-owies32.lock

```

**Comment by adamcstephens** [comment]
@mitchellh Can we reopen this?

**Comment by adamcstephens** [comment]
@mitchellh Can we reopen this?

**Comment by mitchellh** [comment]
@adamcstephens If I remember correctly, these final four weren't worth the effort to remove, hence the "much better" above and not "completely gone." If someone were to open a PR to fix these I'd take a look but I don't think I'd consider 4 leftover files going into the temporary directory a bug, since that directory is made for temporary files anyways that are safe enough to just delete via any other process.

But like I said, if someone pinpoints it and fixes it, I'd gladly merge it, but I distinctly remember this issue making many more files before, doing my best to do cleanups, getting to these last 4 and realizing they'd be really tricky, and just calling it pragmatically good enough.

**Comment by varac** [comment]
i agree that for 4 files we don't need to bother. however, it's *4 files per vagrant call*, which left 2500 (!) files and directories in my /tmp folder over a period of ~3 weeks, and this is quite annoying, because it completely break useful auto-completion and slows down searching for files in /tmp.

**Comment by sni** [comment]
Any news on this?

´´´
%> vagrant --version
Vagrant 1.7.2

%> ls -la /tmp/vagrant2015*
-rw------- 1 build build   0 Jul  6 16:59 /tmp/vagrant20150706-10271-1gc1u9u
-rw-rw-r-- 1 build build   0 Jul  6 16:59 /tmp/vagrant20150706-10271-1gc1u9u1
-rw------- 1 build build   0 Jul  6 16:59 /tmp/vagrant20150706-10271-q770al
-rw-rw-r-- 1 build build 141 Jul  6 16:59 /tmp/vagrant20150706-10271-q770al2

%> cat /tmp/vagrant2015*
source "https://rubygems.org"
source "http://gems.hashicorp.com"
gem "vagrant", "= 1.7.2"
group :plugins do
gem "vagrant-share", nil, {}
end
´´´

Also depending on the the command, ...lock files are created and never removed.

**Comment by remram44** [comment]
This is very not solved; see #6301

**Comment by avdd** [comment]
Nope, still not fixed in 1.8.0, despite what the changelog says
